pallets/asset-registry/src/lib.rs

```diff
@@ -91,9 +91,16 @@ pub mod pallet {
 	#[pallet::genesis_build]
 	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
 		fn build(&self) {
+			// REVIEW: You probably want to make sure the genesis config is sane?
+			assert_ne!(self.core_asset_id, self.next_asset_id, "core asset id needs to be unique");
+			// REVIEW: maybe even:
+			assert!(self.core_asset_id < self.next_asset_id, "core asset id needs to be unique");
 			CoreAssetId::<T>::put(self.core_asset_id);
 			NextAssetId::<T>::put(self.next_asset_id);
+			// REVIEW: Maybe track them in a btree-set? Might be overkill, not sure.
 			self.asset_ids.iter().for_each(|(name, asset_id)| {
+				// REVIEW: maybe:
+				assert!(asset_id < &self.next_asset_id, "asset ids need to be unique");
```

Comment:
Yes, this pallet had the refactor coming for quite some time.

And now, we're completely reworking our asset-registry. It will include better handling of assets in  genesis config too.